autorest/azure/example/main.go

```diff
@@ -104,27 +150,92 @@ func getResourceGroups(client *autorest.Client) (*string, error) {
 		return nil, err
 	}
 
+	value := struct {
+		ResourceGroups []struct {
+			Name string `json:"name"`
+		} `json:"value"`
+	}{}
+
 	defer resp.Body.Close()
-	contents, err := ioutil.ReadAll(resp.Body)
+	dec := json.NewDecoder(resp.Body)
+	err = dec.Decode(&value)
 	if err != nil {
 		return nil, err
 	}
 
-	contentsString := string(contents)
+	var names = make([]string, len(value.ResourceGroups))
+	for i, name := range value.ResourceGroups {
+		names[i] = name.Name
+	}
+	return names, nil
+}
 
-	return &contentsString, nil
+func saveToken(spt azure.Token) {
```

Comment:
What would you expect Refresh() to do with an error here? does it return it all the way out and make Refresh() error (which will kill off your in-flight request) ?